spring-cloud-azure-markdown-appdendix-generator/src/main/java/com/azure/spring/cloud/internal/Main.java

```diff
@@ -0,0 +1,255 @@
+                                + "/" + propertyCount + "] were matching the pattern [" + inclusionPattern + "]");
+                System.out.println("Successfully built the description table");
+                if (names.isEmpty()) {
+                    System.out.println("Will not update the table, since no configuration properties were found!");
+                    return;
+                }
+
+                int[] offset = new int[2];
+                String[] key = new String[1];
+                descriptions.keySet().stream().max(Comparator.comparingInt(String::length))
+                        .ifPresent(longest -> offset[0] = longest.length());
+                names.stream().max(Comparator.comparingInt(aName -> descriptions.get(aName).getDescription().length()))
+                        .ifPresent(longest -> key[0] = longest);
+                offset[1] = descriptions.get(key[0]).getDescription().trim().length();
+                generatePropertiesFiles(outputFile, names, descriptions, offset);
+            }
+            catch (IOException e) {
+                throw new IllegalStateException(e);
+            }
+        }
+
+        /**
+         * Prefix check.
+         */
+        protected boolean resourceNameContainsPattern(Resource resource) {
+            try {
+                return resource.getURL().toString().contains("cloud");
+            }
+            catch (Exception e) {
+                System.out.println("Exception [" + e + "] for resource [" + resource
+                        + "] occurred while trying to retrieve its URL");
+                return false;
+            }
+        }
+
+        /**
+         * Get configuration properties name and descriptions.
+         */
+        protected Resource[] getResources() throws IOException {
+            return new PathMatchingResourcePatternResolver()
+                    .getResources("classpath*:/META-INF/spring-configuration-metadata.json");
+        }
+
+        private String getCurrentDate() {
+            SimpleDateFormat sdf = new SimpleDateFormat();
+            sdf.applyPattern("MM/dd/yyyy");
+            Date date = new Date();
+            return sdf.format(date);
+        }
+
+        private String paddingWithChar(String printableStr, char c, int maxLength) {
+            String trimmedString = printableStr.trim();
+            return printableStr + String.join("", Collections.nCopies(maxLength - trimmedString.length(), c + ""));
+        }
+
+        private String capitalize(String letters) {
+            if ("db".equals(letters) | "jms".equals(letters) | "b2c".equals(letters)) {
+                return letters.toUpperCase();
+            }
+            else {
+                return letters.substring(0, 1).toUpperCase().concat(letters.substring(1));
+            }
+        }
+
+        private String generateAnchorName(String outputFile) {
+            outputFile = outputFile.substring(outputFile.lastIndexOf("/") + 26, outputFile.lastIndexOf("."))
```

Comment:
Don't use `26`, use `"_configuration-properties-".length()` instead.
This requires that `outputFile` should have prefix `_configuration-properties-`, add this requirement into java doc of `Generator#generate`